simpler-wick.tex

```diff
@@ -1,330 +1,87 @@
-%% Page breaks can be avoided at particular places by using \\*
-\allowdisplaybreaks[2]
-
-%% Tables
-%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-
-\usepackage{array}     % New column types, including >{}x<{}
-\usepackage{booktabs}  % Provides nicer horizontal lines
-\usepackage{multirow}  % Allows cells to span multiple rows
-%\usepackage{longtable} % Allows for tables to span multiple pages
-
-%% Define the maths version of clr columns.
-\newcolumntype{C}{>{\(}c<{\)}}
-\newcolumntype{L}{>{\(}l<{\)}}
-\newcolumntype{R}{>{\(}r<{\)}}
-
-%% Other Packages
-%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-
-\usepackage{listings}    % Code listings
-\usepackage{minted}      % Use Pygments
-\usepackage{fp}          % Floating point arithmetics
-\usepackage{makeidx}     % Index creation
-\usepackage{xr}          % Cross-referencing
-
-\usepackage{hyperref}    % Automatically inserts hyperlinks.
-\usepackage{cleveref}    % Use `\cref{}` to reference anything
-
-\usepackage{autonum}
-
-%% Other modifications
-%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-
-% Modify the skip after each paragraph
-\setlength{\parskip}{1ex plus 0.5ex minus 0.2ex}
-\setlength{\parindent}{0pt}
-
-\providecommand\href[2]{\texttt{#1}}
-\providecommand\hypertarget[2]{\texttt{#1}}
-\providecommand\hyperlink[2]{\texttt{#1}}
-
-\newenvironment{example}
-  {\VerbatimEnvironment
-   \begin{VerbatimOut}{example.out}}
-  {\end{VerbatimOut}%
-   \vspace{1ex}%
-   \setlength{\parindent}{0pt}%
-   \fbox{\begin{minipage}{0.5\linewidth}%
-     \inputminted[resetmargins]{latex}{example.out}%
-   \end{minipage}%
-   \hspace{0.05\linewidth}%
-   \begin{minipage}{0.43\linewidth}%
-     \input{example.out}%
-   \end{minipage}%
-   \vspace{1ex}}}
-
-\let\origtexttt=\texttt
-\def\texttt#1{{\def\textunderscore{\char`\_}\def\textbraceleft{\char`\{}\def\textbraceright{\char`\}}\origtexttt{#1}}}
-\def\exclamationmarktext{!}
-\def\atmarktext{@}
-
-{
-  \catcode`\|=12
-  \gdef\pgfmanualnormalbar{|}
-  \catcode`\|=13
-  \AtBeginDocument{\gdef|{\ifmmode\pgfmanualnormalbar\else\expandafter\verb\expandafter|\fi}}
-}
-
-\makeatother
-
-\makeindex
-
-%% Define some slightly nicer colors
-\definecolor{link-color}{RGB}{96 0 0}
-\definecolor{cite-color}{RGB}{0 96 0}
-\definecolor{file-color}{RGB}{0 0 96}
-\definecolor{url-color}{RGB}{0 0 96}
-\definecolor{link-border-color}{RGB}{255 159 159}
-\definecolor{cite-border-color}{RGB}{159 255 159}
-\definecolor{file-border-color}{RGB}{159 159 255}
-\definecolor{url-border-color}{RGB}{159 159 255}
-
-\hypersetup{
-  pdfborder=0 0 0,
+\input tikz.tex
+
+% makeatletter
+\edef\simplerwickatcode{\the\catcode`\@}
+\catcode`\@=11
+ 
+% Definition of \@ifnextchar from LaTeX
+\long\def\@ifnextchar#1#2#3{%
+  \let\reserved@d=#1%
+  \def\reserved@a{#2}%
+  \def\reserved@b{#3}%
+  \futurelet\@let@token\@ifnch}
+ 
+\def\@ifnch{%
```

Comment:
What is the purpose of this `\@ifnch`?  Could you add a small comment describing it?

Reply:
`\@ifnch` is a helper macro for `\@ifnextchar`.  The frontend `\@ifnextchar` only saves the arguments and scans for the next token.  Then it hands over to `\@ifnch` to before the branching.  This has to be done like this because of the way `\futurelet` works.

Reply:
Thanks for the clarification.